src/fsipy/automatedPostprocessing/postprocessing_h5py/spectrograms.py

```diff
@@ -0,0 +1,778 @@
+                             "meters).")
+    parser.add_argument('--y-sphere', type=float, default=0.0,
+                        help="Y-coordinate of the center of the sphere used to include points for spectrogram (in "
+                             "meters).")
+    parser.add_argument('--z-sphere', type=float, default=0.0,
+                        help="Z-coordinate of the center of the sphere used to include points for spectrogram (in "
+                             "meters).")
+    parser.add_argument('--dvp', type=str, default="v",
+                        help="Quantity to postprocess. Choose 'v' for velocity, 'd' for displacement, 'p' for "
+                             "pressure, or 'wss' for wall shear stress.")
+    parser.add_argument('--Re_a', type=float, default=0.0,
+                        help="Assuming linearly increasing Reynolds number: Re(t) = Re_a*t + Re_b. If both Re_a and "
+                             "Re_b are 0, the plot won't be against Reynolds number.")
+    parser.add_argument('--Re_b', type=float, default=0.0,
+                        help="Assuming linearly increasing Reynolds number: Re(t) = Re_a*t + Re_b. If both Re_a and "
+                             "Re_b are 0, the plot won't be against Reynolds number.")
+    parser.add_argument('--interface-only', action='store_true',
+                        help="Generate spectrogram only for the fluid-solid interface. If present, interface-only "
+                             "spectrogram will be generated; otherwise, the volumetric spectrogram will include all "
+                             "fluid in the sac or all nodes through the wall.")
+    parser.add_argument('--component', type=str, default="mag",
+                        help="Component of the data to visualize. Choose 'x', 'y', 'z', or 'mag' (magnitude).")
+    parser.add_argument('--sampling-method', type=str, default="RandomPoint",
+                        help="Sampling method for spectrogram generation. Choose from 'RandomPoint' (random nodes), "
+                             "'SinglePoint' (single point specified by 'point_id'), or 'Spatial' (ensures uniform "
+                             "spatial sampling, e.g., in the case of fluid boundary layer, the sampling will not bias "
+                             "towards the boundary layer).")
+    parser.add_argument('--n-samples', type=int, default=10000,
+                        help="Number of samples to generate spectrogram data (ignored for SinglePoint sampling).")
+    parser.add_argument('--point-id', type=int, default=-1000000,
+                        help="Point ID for SinglePoint sampling. Ignored for other sampling methods.")
+
+    parser.add_argument('--overlap-frac', type=float, default=0.75,
+                        help="Fraction of overlap between adjacent windows.")
+    parser.add_argument('--window', type=str, default="blackmanharris",
+                        help="Window function to be used for spectrogram computation.")
```

Comment:
Not important, but we could explanation of what other window functions are available. Here https://docs.scipy.org/doc/scipy/reference/signal.windows.html scipy lists possible window functions, so it might be nice to add this as a reference point.